tests/lti/conftest.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-@pytest.fixture
+
+from __future__ import unicode_literals
+
+import functools
+
+import mock
+import pytest
+from pyramid import testing
+from pyramid.request import apply_request_extensions
+
+from lti.models import AuthData
+
+
+def autopatcher(request, target, **kwargs):
+    """Patch and cleanup automatically. Wraps :py:func:`mock.patch`."""
+    options = {'autospec': True}
+    options.update(kwargs)
+    patcher = mock.patch(target, **options)
+    obj = patcher.start()
+    request.addfinalizer(patcher.stop)
+    return obj
+
+
+@pytest.fixture
+def patch(request):
+    return functools.partial(autopatcher, request)
+
+
+@pytest.fixture
+def pyramid_request():
+    """
+    Return a dummy Pyramid request object.
+
+    This is the same dummy request object as is used by the pyramid_config
+    fixture below.
+
+    """
+    pyramid_request = testing.DummyRequest()
+
+    pyramid_request.auth_data = mock.create_autospec(AuthData, instance=True)
+    pyramid_request.auth_data.get_canvas_server.return_value = 'https://TEST_CANVAS_SERVER.com'
+    pyramid_request.auth_data.get_lti_secret.return_value = 'TEST_CLIENT_SECRET'
+
+    return pyramid_request
```

Comment:
This is a new fixture that tests can use whenever they need a test Pyramid `Request` object.